### ISSUE TITLE
Varargs not allowed in function types

### ISSUE THREAD

**Issue by ds84182** [rust-analyzer/rust-analyzer#1214]
You can reproduce with this code:
```rust
struct Foo {
    pub bar: unsafe extern "C" fn(a: i32, b: f64, ...) -> bool,
}
```

Currently gives a bunch of syntax errors.

**Comment by edwin0cheng** [comment]
It should be fixed in #1197 , would you mind tell me which commit you are using ? I just tested it in master, it has no error in SyntaxTree :
```
SOURCE_FILE@[0; 78)
  STRUCT_DEF@[0; 78)
    STRUCT_KW@[0; 6) "struct"
    WHITESPACE@[6; 7) " "
    NAME@[7; 10)
      IDENT@[7; 10) "Foo"
    WHITESPACE@[10; 11) " "
    NAMED_FIELD_DEF_LIST@[11; 78)
      L_CURLY@[11; 12) "{"
      WHITESPACE@[12; 17) "\n    "
      NAMED_FIELD_DEF@[17; 75)
        VISIBILITY@[17; 20)
          PUB_KW@[17; 20) "pub"
        WHITESPACE@[20; 21) " "
        NAME@[21; 24)
          IDENT@[21; 24) "bar"
        COLON@[24; 25) ":"
        WHITESPACE@[25; 26) " "
        FN_POINTER_TYPE@[26; 75)
          UNSAFE_KW@[26; 32) "unsafe"
          WHITESPACE@[32; 33) " "
          ABI@[33; 43)
            EXTERN_KW@[33; 39) "extern"
            WHITESPACE@[39; 40) " "
            STRING@[40; 43) "\"C\""
          WHITESPACE@[43; 44) " "
          FN_KW@[44; 46) "fn"
          PARAM_LIST@[46; 67)
            L_PAREN@[46; 47) "("
            PARAM@[47; 53)
              BIND_PAT@[47; 48)
                NAME@[47; 48)
                  IDENT@[47; 48) "a"
              COLON@[48; 49) ":"
              WHITESPACE@[49; 50) " "
              PATH_TYPE@[50; 53)
                PATH@[50; 53)
                  PATH_SEGMENT@[50; 53)
                    NAME_REF@[50; 53)
                      IDENT@[50; 53) "i32"
            COMMA@[53; 54) ","
            WHITESPACE@[54; 55) " "
            PARAM@[55; 61)
              BIND_PAT@[55; 56)
                NAME@[55; 56)
                  IDENT@[55; 56) "b"
              COLON@[56; 57) ":"
              WHITESPACE@[57; 58) " "
              PATH_TYPE@[58; 61)
                PATH@[58; 61)
                  PATH_SEGMENT@[58; 61)
                    NAME_REF@[58; 61)
                      IDENT@[58; 61) "f64"
            COMMA@[61; 62) ","
            WHITESPACE@[62; 63) " "
            DOTDOTDOT@[63; 66) "..."
            R_PAREN@[66; 67) ")"
          WHITESPACE@[67; 68) " "
          RET_TYPE@[68; 75)
            THIN_ARROW@[68; 70) "->"
            WHITESPACE@[70; 71) " "
            PATH_TYPE@[71; 75)
              PATH@[71; 75)
                PATH_SEGMENT@[71; 75)
                  NAME_REF@[71; 75)
                    IDENT@[71; 75) "bool"
      COMMA@[75; 76) ","
      WHITESPACE@[76; 77) "\n"
      R_CURLY@[77; 78) "}"
```

**Comment by ds84182** [comment]
Ah, that was it. I was trying to use GitHub search to figure out if support had been added, but nothing came up. Thanks!
